scala-package/core/src/main/scala/org/apache/mxnet/IO.scala

```diff
@@ -170,6 +180,10 @@ object DataBatch {
     private var label: IndexedSeq[NDArray] = null
     private var index: IndexedSeq[Long] = null
     private var pad: Int = 0
+    private var dataLayout: String = "NCHW"
```

Comment:
For layouts I suggest to use extensible Enums similar to this https://blogs.oracle.com/darcy/mixing-in-an-enum.  
free floating strings are generally not a good practice because it removes compile time type-checks. 
WDYT? does it make sense?